File: ProjectEuler4/main.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <chrono>


bool isPalindrom(int potentialPalindrom);

int main(int argc, char const *argv[])
{
    std::chrono::time_point<std::chrono::system_clock> start, end;
    start = std::chrono::system_clock::now();

    int largestPalindrom = 0;

    for (int i = 100; i < 1000; ++i)
    {
        for (int j = i; j < 1000; ++j)
        {
            if(isPalindrom(i*j) && i*j > largestPalindrom) largestPalindrom = i*j;
        }
    }

    end = std::chrono::system_clock::now();
    long elapsed_millis = std::chrono::duration_cast<std::chrono::milliseconds>(end-start).count();
    std::cout << "Elapsed Milliseconds: " << elapsed_millis << std::endl;

    std::cout << largestPalindrom << std::endl;
    return 0;
}

bool isPalindrom(int potentialPalindrom)
{
    auto potentialPalindromString = std::to_string(potentialPalindrom);
    auto reversedString = potentialPalindromString;
    std::reverse(reversedString.begin(), reversedString.end());

    return potentialPalindromString == reversedString;
}

File: ProjectEuler4/src/main.rs
use std::time::{Duration, Instant};

fn is_palindrom(potential_palindrom: i32) -> bool{
    let potential_palindrom_string = potential_palindrom.to_string();

    for (c1, c2) in potential_palindrom_string.chars().zip(potential_palindrom_string.chars().rev()) {
        if c1 != c2 {
            return false;
        }
    }

    return true;
}

fn main() {
    let now = Instant::now();

    let mut largest_palindrom = 0;

    for i in 100..1000 {
        for j in 100..1000 {
            if is_palindrom(i * j) && i * j > largest_palindrom{
                largest_palindrom = i * j;
            }
        }
    }

    println!("Largest Palindrom: {}", largest_palindrom);
    println!("Elapsed µs: {}", now.elapsed().as_micros());
}
